plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -727,17 +724,18 @@ func PodFitsHostPorts(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.N
 		wantPorts = getUsedPorts(pod)
 	}
 	if len(wantPorts) == 0 {
-		return true, nil
+		return true, nil, nil
 	}
 
 	// TODO: Aggregate it at the NodeInfo level.
 	existingPorts := getUsedPorts(nodeInfo.Pods()...)
+	var predicateFails []algorithm.PredicateFailureReason
 	for wport := range wantPorts {
 		if wport != 0 && existingPorts[wport] {
-			return false, ErrPodNotFitsHostPorts
+			predicateFails = append(predicateFails, ErrPodNotFitsHostPorts)
```

Comment:
Well - but currently it's not possible, and it doesn't seem that it will happen any time soon. So please make a return here (it will be simple to change in the future if we decide to do so).